crates/query-engine/translation/src/translation/query/values.rs

```diff
@@ -84,48 +84,88 @@ fn type_to_ast_scalar_type(typ: &database::Type) -> sql::ast::ScalarType {
 
 /// Convert a variable into a SQL value.
 pub fn translate_variable(
+    state: &mut State,
     variables_table: sql::ast::TableReference,
     variable: String,
     r#type: &database::Type,
 ) -> sql::ast::Expression {
-    let variables_reference =
-        Expression::ColumnReference(sql::ast::ColumnReference::AliasedColumn {
-            table: variables_table,
-            column: sql::helpers::make_column_alias(sql::helpers::VARIABLES_FIELD.to_string()),
-        });
+    let variables_reference = Expression::ColumnReference(ColumnReference::AliasedColumn {
+        table: variables_table,
+        column: sql::helpers::make_column_alias(sql::helpers::VARIABLES_FIELD.to_string()),
+    });
 
-    match r#type {
-        database::Type::CompositeType(_type_name) => {
-            let exp = sql::ast::Expression::BinaryOperation {
+    let projected_variable_exp = match r#type {
+        database::Type::CompositeType(_) | database::Type::ArrayType(_) => {
```

Comment:
Could you please add a comment of the intention of the difference between this block and the next block? Something along the lines of:

```suggestion
        // For composite types and array types, we use `->` to return the value as `json`
        // because ...
        database::Type::CompositeType(_) | database::Type::ArrayType(_) => {
```

Reply:
Sensible! I'll do that.

Reply:
These parts should be better commented now.

I also simplified the process by only using the `->` operator and moving the textual extraction to the `ScalarType`-case, which means that the function only assumes that th expression it's working on is a `jsonb`-value, and not contextually dependent on whether it's translating a scalar type or composite/array type.